### ISSUE TITLE
ch-fromhost should put injected files at same path if no inference and no --dest

### ISSUE THREAD

**Issue by reidpr** [hpc/charliecloud#322]
(Per @samcmill in PR #321.)

Current behavior:

```
$ ch-fromhost -v -b /etc/resolv.conf ex3/rootfs
injecting into image: ex3/rootfs
  /etc/resolv.conf -> 
ch-fromhost: no destination for: /etc/resolv.conf
```

Desired behavior:

```
$ ch-fromhost -v -b /etc/resolv.conf ex3/rootfs 
injecting into image: ex3/rootfs
  /etc/resolv.conf -> /etc
not running ldconfig
```

**Comment by reidpr** [comment]
@samcmill, `/etc/resolv.conf` should be bind-mounted by default already. Is that not working? Do you need this feature for other files as well?

**Comment by samcmill** [comment]
The specific issue is that CentOS images do not have `/etc/resolv.conf`, so the bind mount fails (#320).  An additional extension might be `ch-fromhost --bind-mounts` (or similar) that would touch every required bind mount point in the image to ensure they exist.  This could be worth a new, separate command since no files are actually copied from the host?

This specific issue is just an illustration of the proposed feature.  It also has uses outside this specific case.

**Comment by reidpr** [comment]
Thanks for clarifying.

Interestingly, my CentOS images have that file and work fine. Good to know that can't be relied on.

**Comment by samcmill** [comment]
I assume you are using a Docker workflow to get your CentOS images?

For the same `centos:7` image from Docker Hub, the `ch-docker2tar` workflow produces a tarfile with `/etc/resolv.conf`, but the `skopeo` + `umoci` workflow does not.  I assume this is because `ch-docker2tar` invokes `docker create`, which is likely modifying the image.  A tar file generated by `docker save` does not include `/etc/resolv.conf`.

Since unprivileged containers is one of the key differentiating features of Charliecloud, it would be very nice to support the `skopeo` + `umoci` workflow.

**Comment by reidpr** [comment]
Yes, that's correct. We want `skopeo` + `umoci` to work. Do you have suggestions for a basic image or workflow that would be good to put in the test suite?

**Comment by samcmill** [comment]
Please see #325.  The `centos:7` image from Docker Hub would be a good test case since it exhibits some differences relative to the existing `ch-docker2tar` workflow.

**Comment by reidpr** [comment]
@samcmill, can you say a little more about the "uses outside this specific case"?

I realized that "same path" gets difficult unless the injected file has an absolute path and the host and guest distributions are sufficiently similar in how they organize things. I'd like to understand things better before adding this, to make sure we do it right.

**Comment by samcmill** [comment]
For instance, I have been playing around with injecting the host MPI into containers.  For InfiniBand support, in addition to the usual suspects like `/lib64/libibverbs.so.1`, I found I also needed to inject things like `/etc/libibverbs.d` (directory) and `/etc/dat.conf` (file).

**Comment by reidpr** [comment]
OK, thank you.

**Comment by reidpr** [comment]
We discussed this at today's team meeting.

* Our current opinion on best practices is that it's better to build a portable MPI into the container, rather than injecting it at run time, though we do want to understand better situations where that doesn't seem to work (either so we can help fix them or to challenge our assumptions). We also added a new issue (#379) to write an FAQ on MPI best practices.

* A workaround does exist (`--dest`).

* We want `ch-fromhost` to go away, instead folding the features into `ch-run`; thus, we're hesitant to add new features to it.

For these reasons, we're going to defer this issue and revisit it in the future (maybe when `ch-fromhost` is gone). Opinions welcome of course.
